### PR TITLE
fix: Create host fails on 2.5.2

### LINKED ISSUE #207: Create host fails on 2.5.2
**Describe the bug**
Creating host cannot load nic interface form. 

**Versions:**
Foreman: 2.5.2
foreman_fog_proxmox: 0.14.0
fog-proxmox: 0.14.

**To Reproduce**
1. Go to 'Create Host'
2. Select "Deploy on" to a Proxmox cluster
3. Interface tab header turns red with the error `Error loading interfaces information: Internal Server Error`

**Backtrace and error attachached:**
[fog_proxmox_backtrace.txt](https://github.com/theforeman/foreman_fog_proxmox/files/6887391/fog_proxmox_backtrace.txt)

### PR DESCRIPTION
**Pull request by tristanrobert**
Fixes #207

Replace https://github.com/theforeman/foreman_fog_proxmox/pull/239

### CODE REVIEW

File: app/helpers/proxmox_vm_volumes_helper.rb [hunk 1]
```diff
@@ -52,12 +49,14 @@ def parsed_typed_volumes(args, type, parsed_vm)
...
   def parse_hard_disk_volume(args)
+    logger.debug(format(_('parse_hard_disk_volume(): args=%<args>s'), args: args))
     disk = {}
     disk[:id] = args['id'] if args.key?('id')
     disk[:volid] = args['volid'] if args.key?('volid')
     disk[:storage] = args['storage'].to_s if args.key?('storage')
     disk[:size] = args['size'].to_i if args.key?('size')
-    add_disk_options(disk, args)
+    add_disk_options(disk, args) unless args.key?('options')
+    disk[:options] = args['options'] if args.key?('options')
```
Comment (ekohl) [inline app/helpers/proxmox_vm_volumes_helper.rb:59]: I'd reduce it to a simple if/else.

File: app/models/foreman_fog_proxmox/proxmox_volumes.rb [hunk 2]
```diff
@@ -59,9 +59,9 @@ def update_cdrom(vm, disk, volume_attributes)
     end
 
     def extend_volume(vm, id, diff_size)
-      extension = '+' + (diff_size / GIGA).to_s + 'G'
-      logger.info(format(_('vm %<vmid>s extend volume %<volume_id>s to %<extension>s'), vmid: vm.identity,
-volume_id: id, extension: extension))
+      extension = '+' + diff_size.to_s + 'G'
+      logger.info(format(_('vm %<vmid>s extend volume %<volume_id>s to %<extension>s'),
```
Comment (ekohl) [inline app/models/foreman_fog_proxmox/proxmox_volumes.rb:63]: I realize it's not part of this PR, logger statements shouldn't be translated.

File: app/models/foreman_fog_proxmox/proxmox_volumes.rb [hunk 3]
```diff
@@ -59,9 +59,9 @@ def update_cdrom(vm, disk, volume_attributes)
     end
 
     def extend_volume(vm, id, diff_size)
-      extension = '+' + (diff_size / GIGA).to_s + 'G'
-      logger.info(format(_('vm %<vmid>s extend volume %<volume_id>s to %<extension>s'), vmid: vm.identity,
-volume_id: id, extension: extension))
+      extension = '+' + diff_size.to_s + 'G'
```
Comment (ekohl) [inline app/models/foreman_fog_proxmox/proxmox_volumes.rb:62]: Also partly unrelated, but usually this is done using string interpolation:
```suggestion
      extension = "+#{diff_size}G"
```

File: app/models/foreman_fog_proxmox/proxmox_vm_queries.rb [hunk 4]
```diff
@@ -61,10 +61,9 @@ def find_vm_by_uuid(uuid)
       nodes.each do |node|
         vm = find_vm_in_servers_by_vmid(node.servers, vmid)
         vm ||= find_vm_in_servers_by_vmid(node.containers, vmid)
-        unless vm.nil?
-          logger.debug("found vm #{vmid} on node #{node.node}")
-          break
-        end
+        next if vm.nil?
+        logger.debug("found vm #{vmid} on node #{node.node}")
+        break
```
Comment (ekohl) [inline app/models/foreman_fog_proxmox/proxmox_vm_queries.rb:66]: You can write this method using [`find`](https://ruby-doc.org/core-2.7.1/Enumerable.html#method-i-find):
```ruby
def find_vm_by_uuid(uuid)
  vmid = extract_vmid(uuid)
  nodes.find do |node|
    find_vm_in_servers_by_vmid(node.servers, vmid) || find_vm_in_servers_by_vmid(node.containers, vmid)
  end
end
```

**Review by ekohl — COMMENTED** [review]
I'd consider making a separate PR for the translations. There is a lot going on in this PR and that generally makes it harder to review & merge.

Personally I'd start with a PR that removes all translations from logger statements, then fix this and then a PR that adds translations.

File: app/models/foreman_fog_proxmox/proxmox_volumes.rb (hunk 2 above)
Comment (tristanrobert) [reply app/models/foreman_fog_proxmox/proxmox_volumes.rb:63]: I will see if theforeman-rubocop rules accept this style

File: app/helpers/proxmox_vm_volumes_helper.rb (hunk 1 above)
Comment (tristanrobert) [reply app/helpers/proxmox_vm_volumes_helper.rb:59]: I will see if theforeman-rubocop rules accept this style

File: app/helpers/proxmox_vm_volumes_helper.rb [hunk 5]
```diff
@@ -52,12 +49,14 @@ def parsed_typed_volumes(args, type, parsed_vm)
   end
 
   def parse_hard_disk_volume(args)
+    logger.debug(format(_('parse_hard_disk_volume(): args=%<args>s'), args: args))
```
[the remark this answers is not in the document]
Comment (ekohl) [reply app/helpers/proxmox_vm_volumes_helper.rb:52]: Agreed: I suggested the same in https://github.com/theforeman/foreman_fog_proxmox/pull/240#pullrequestreview-1206760347.

File: app/models/foreman_fog_proxmox/proxmox_vm_queries.rb (hunk 4 above)
Comment (ekohl) [reply app/models/foreman_fog_proxmox/proxmox_vm_queries.rb:66]: I just realized after looking: my suggestion returns the node, not the found VM so I think you can ignore my suggestion.
